Mathlib/Probability/Kernel/Disintegration/MeasurableStieltjes.lean

```diff
@@ -0,0 +1,494 @@
+      tendsto_const_nhds ?_ (stieltjesFunction_le_one hf a)
+  intro x
+  rw [Function.comp_apply, ← stieltjesFunction_eq hf]
+  exact (hf.stieltjesFunction a).mono (le_of_lt (h_exists x).choose_spec.2)
+
+lemma IsRatCDF.measurable_stieltjesFunction (x : ℝ) :
+    Measurable fun a ↦ hf.stieltjesFunction a x := by
+  have : (fun a ↦ hf.stieltjesFunction a x) = fun a ↦ ⨅ r : { r' : ℚ // x < r' }, f a ↑r := by
+    ext1 a
+    rw [← StieltjesFunction.iInf_rat_gt_eq]
+    congr with q
+    rw [stieltjesFunction_eq]
+  rw [this]
+  exact measurable_iInf (fun q ↦ hf.measurable.eval)
+
+lemma IsRatCDF.stronglyMeasurable_stieltjesFunction (x : ℝ) :
+    StronglyMeasurable fun a ↦ hf.stieltjesFunction a x :=
+  (measurable_stieltjesFunction hf x).stronglyMeasurable
+
+section Measure
+
+lemma IsRatCDF.measure_stieltjesFunction_Iic (a : α) (x : ℝ) :
+    (hf.stieltjesFunction a).measure (Iic x) = ENNReal.ofReal (hf.stieltjesFunction a x) := by
+  rw [← sub_zero (hf.stieltjesFunction a x)]
+  exact (hf.stieltjesFunction a).measure_Iic (tendsto_stieltjesFunction_atBot hf a) _
+
+lemma IsRatCDF.measure_stieltjesFunction_univ (a : α) :
+    (hf.stieltjesFunction a).measure univ = 1 := by
+  rw [← ENNReal.ofReal_one, ← sub_zero (1 : ℝ)]
+  exact StieltjesFunction.measure_univ _ (tendsto_stieltjesFunction_atBot hf a)
+    (tendsto_stieltjesFunction_atTop hf a)
+
+instance IsRatCDF.instIsProbabilityMeasure_stieltjesFunction (a : α) :
+    IsProbabilityMeasure (hf.stieltjesFunction a).measure :=
+  ⟨measure_stieltjesFunction_univ hf a⟩
+
+lemma IsRatCDF.measurable_measure_stieltjesFunction :
+    Measurable fun a ↦ (hf.stieltjesFunction a).measure := by
+  rw [Measure.measurable_measure]
+  refine fun s hs ↦ MeasurableSpace.induction_on_inter
+    (C := fun s ↦ Measurable fun b ↦ StieltjesFunction.measure (hf.stieltjesFunction b) s)
+    (borel_eq_generateFrom_Iic ℝ) isPiSystem_Iic ?_ ?_ ?_ ?_ hs
+  · simp only [measure_empty, measurable_const]
+  · rintro S ⟨u, rfl⟩
+    simp_rw [measure_stieltjesFunction_Iic hf _ u]
+    exact (measurable_stieltjesFunction hf u).ennreal_ofReal
+  · intro t ht ht_cd_meas
+    have : (fun a ↦ (hf.stieltjesFunction a).measure tᶜ) =
+        (fun a ↦ (hf.stieltjesFunction a).measure univ)
+          - fun a ↦ (hf.stieltjesFunction a).measure t := by
+      ext1 a
+      rw [measure_compl ht (measure_ne_top (hf.stieltjesFunction a).measure _), Pi.sub_apply]
+    simp_rw [this, measure_stieltjesFunction_univ hf]
+    exact Measurable.sub measurable_const ht_cd_meas
+  · intro f hf_disj hf_meas hf_cd_meas
+    simp_rw [measure_iUnion hf_disj hf_meas]
+    exact Measurable.ennreal_tsum hf_cd_meas
+
+end Measure
+
+end IsRatCDF.stieltjesFunction
+
+section stieltjesOfMeasurableRat
+
+variable {f : α → ℚ → ℝ}
+
+/-- Turn a measurable function `f : α → ℚ → ℝ` into a measurable function `α → StieltjesFunction`.
+Composition of `toRatCDF` and `IsRatCDF.stieltjesFunction`. -/
```

Comment:
Could you emphasize in the docstring that, at a point `a` where `f a` is not the restriction of a Stieltjes function going from 0 at -infinity to 1 at +infinity, then `f a` is discarded and replaced by an arbitary choice?